### PR TITLE
289 substance disp name regex

### LINKED ISSUE #289: substance name regexp accepts "*" and "_"
As a *curator*
I want *additional characters to be acceptable in names*
So that *I can properly name things when registering them*

Acceptance Criteria:
For both the display and preferred name
- Accept an asterisk "*"
- Accept an underscore "_"

Test:

### PR DESCRIPTION
**Pull request by stevenhausman-gdit**
Closes #289 

Added * and _ to display name & preferred name.
Added new characters to testing

### CODE REVIEW

File: chemreg/substance/models.py [hunk 1]
```diff
@@ -70,9 +70,9 @@ class Substance(CommonInfo):
         substance_histories (QuerySet): One to Many Substance history resources (not implemented yet)
     """
 
-    preferred_name_regex = "^[a-zA-Z0-9 =<>\\-':.,^%&/{}[\\]()+?=]{3,}$"
-    display_name_regex = "^[a-zA-Z0-9 =<>\\-':.,^%&/{}[\\]()+?=]{3,}$"
-    casrn_regex = "^[0-9]{2,7}-[0-9]{2}-[0-9]$"
+    preferred_name_regex = r"^[a-zA-Z0-9 =<>\-\*_':.,^%&\/{}[\]]{3,}$"
+    display_name_regex = r"^[a-zA-Z0-9 =<>\-\*_':.,^%&\/{}[\]]{3,}$"
```
Comment (michael-barbour) [inline chemreg/substance/models.py:74]: Looks like you deleted a chunk of the regex.  When I run `pytest --create-db` the tests fail due to not accepting the `?` character. 

`()+?=`

I like the edits to turn these strings into python regex strings and eliminating the need for double backslashes.

**Review by michael-barbour — CHANGES_REQUESTED** [review]

File: chemreg/substance/models.py [hunk 2]
```diff
@@ -70,9 +70,9 @@ class Substance(CommonInfo):
...
+    casrn_regex = r"^[0-9]{2,7}-[0-9]{2}-[0-9]$"
```
Comment (michael-barbour) [inline chemreg/substance/models.py:75]: Looks like you deleted a chunk of the regex. When I run pytest --create-db the tests fail due to not accepting the ? character.

()+?=

I like the edits to turn these strings into python regex strings and eliminating the need for double backslashes.

**Review by michael-barbour — CHANGES_REQUESTED** [review]

File: chemreg/substance/models.py (hunk 2 above)
Comment (stevenhausman-gdit) [reply chemreg/substance/models.py:75]: Thanks, mb, "=" is already captured. Will fix, adding back "()+?"

**Review by michael-barbour — APPROVED** [review]
These changes looks good.  Tests pass.

There are some no-impact migrations that are required due to the validation changes.  I'm adding them as they don't change anything.  You can +1 this before I merge.

**Review by stevenhausman-gdit — COMMENTED** [review]
Looks fine. Interesting that makemigrations seems to "re-stringify" the validation regex. The output error message would then include escape characters.

**Pull request merged**